plutus-core/plutus-core/src/PlutusCore/Constant/Meaning.hs

```diff
@@ -288,15 +288,46 @@ instance
       , HandleHolesGo j k term holes
       ) => HandleHolesGo i k term (hole ': holes)
 
+-- | If the outermost constructor of the second argument is known and happens to be one of the
+-- constructors of the list data type, then the second argument is returned back. Otherwise the
+-- first one is returned, which is meant to be a custom type error.
+type ThrowOnStuckList :: forall a. [a] -> [a] -> [a]
+type family ThrowOnStuckList err xs where
+    ThrowOnStuckList _   '[]       = '[]
+    ThrowOnStuckList _   (x ': xs) = x ': xs
+    ThrowOnStuckList err _         = err
```

Comment:
I'm kind of horrified that this works :sweat_smile:

Reply:
Yeah, I understand that, it's just kind of bizarre.

Reply:
Not as bizarre as the fact that `ThrowOnStuckList` can actually return the first argument if the second one is neither stuck nor a constructor of `[]`, but that's Haskell's type-level Python.